src/track-promise.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SRC_TRACK_PROMISE_H_
+#define SRC_TRACK_PROMISE_H_
+
+#include "v8.h"
+
+namespace node {
+
+class Environment;
+
+class TrackPromise {
+ public:
+  TrackPromise(v8::Isolate* isolate, v8::Local<v8::Object> object);
```

Comment:
Dumb question, but isn't pass by reference preferable to pass by pointer here? 

Similarly, wouldn't we want to pass the `Local<v8::Object>` by reference? Hardly a big deal though as you can pass `Handle`s and copy them around IIRC.